### PR TITLE
21 update country user models

### PR DESCRIPTION
**Pull request by cherchezlafemme**
https://trello.com/c/jYeRiqG6/21-update-country-and-user-model-in-rails-to-reflect-new-associations

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -2,4 +2,7 @@ class User < ApplicationRecord
   validates_presence_of :name, :email, :password
   has_many :reviews
   has_many :notes
+  has_many :created_sodas, :class_name => 'Soda', :foreign_key => 'creator_id'
+  has_many :last_edited_sodas, :class_name => 'Soda', :foreign_key => 'last_editor_id'
```
Comment (bambery) [inline app/models/user.rb]: Can you provide a use case where we would want this information from the user's side? I thought about it, and couldn't come up with one, so I purposefully omitted this side of the relationship. Should a user care about the last new soda they added into the db? Or the last one they edited? We definitely need it from the soda side, but a user doesn't care

File: app/models/soda.rb [hunk 2]
```diff
@@ -5,7 +5,7 @@ class Soda < ApplicationRecord
   has_many :photos
   has_and_belongs_to_many :flavors
   has_and_belongs_to_many :ingredients
-  has_and_belongs_to_many :notes
+  has_many :notes
```
Comment (bambery) [inline app/models/soda.rb:8]: oops, good catch

File: app/models/user.rb (hunk 1 above)
Comment (cherchezlafemme) [reply app/models/user.rb]: I don't think the user should care. 
Hm... would it work though, if you have the relationship declared only on one side?

Comment (cherchezlafemme) [reply app/models/user.rb]: If it will work without it I can delete this.

**Pull request merged**